r/src/compute-exec.cpp

```diff
@@ -226,31 +226,27 @@ std::shared_ptr<compute::ExecNode> ExecNode_Project(
 // [[arrow::export]]
 std::shared_ptr<compute::ExecNode> ExecNode_Aggregate(
     const std::shared_ptr<compute::ExecNode>& input, cpp11::list options,
-    std::vector<std::string> target_names, std::vector<std::string> out_field_names,
     std::vector<std::string> key_names) {
   std::vector<arrow::compute::internal::Aggregate> aggregates;
   std::vector<std::shared_ptr<arrow::compute::FunctionOptions>> keep_alives;
-
   for (cpp11::list name_opts : options) {
-    auto name = cpp11::as_cpp<std::string>(name_opts[0]);
-    auto opts = make_compute_options(name, name_opts[1]);
+    auto function = cpp11::as_cpp<std::string>(name_opts[0]);
+    auto opts = make_compute_options(function, name_opts[1]);
+    auto target = cpp11::as_cpp<std::string>(name_opts[2]);
+    auto name = cpp11::as_cpp<std::string>(name_opts[3]);
```

Comment:
I think we can access these by name instead of position

```suggestion
    auto function = cpp11::as_cpp<std::string>(name_opts["fun"]);
    auto opts = make_compute_options(function, name_opts["options"]);
    auto target = cpp11::as_cpp<std::string>(name_opts["target"]);
    auto name = cpp11::as_cpp<std::string>(name_opts["name"]);
```